hutch_python/happi.py

```diff
@@ -36,32 +50,63 @@ def get_happi_objs(db, hutch):
         A mapping from item name to item
     """
     # Load the happi Client
-    client = happi.Client(path=db)
-    containers = list()
-    if beamlines is None:
-        beamline_conf = []
+    if None not in (light_ctrl, lightpath):
+        client = light_ctrl.client
     else:
-        # Find upstream items based on lightpath configuration
-        beamline_conf = beamlines.get(hutch.upper())
-    # Something strange is happening if there are no upstream items
-    if not beamline_conf:
-        logger.warning("Unable to find lightpath for %s",
-                       hutch.upper())
-        beamline_conf = {}
-    # Add the complete hutch beamline
-    beamline_conf[hutch.upper()] = {}
-    # Base beamline
-    for beamline, conf in beamline_conf.items():
+        client = happi.Client(path=db)
+    containers = list()
+
+    if light_ctrl is None:
+        # lightpath was unavailable, search by beamline name
+        reqs = dict(beamline=endstation.upper(), active=True)
+        results = client.search(**reqs)
+        containers.extend(res.item for res in results)
+        return _load_devices(*containers)
+
+    # if lightpath exists, we can grab upstream devices
+    dev_names = set()
+    paths = light_ctrl.beamlines[endstation.upper()]
```

Comment:
`tstpython`'s error:
```
2022-10-25 15:17:20 - PID 3751      controller.py: 159 load_beamline      WARNING  - Unable to find TST as a configured endstation, assuming this is an invalid path
2022-10-25 15:17:20 - PID 3751           utils.py: 66  safe_load          ERROR    - Failed to load database after 0.19 s
2022-10-25 15:17:20 - PID 3751           utils.py: 67  safe_load          DEBUG    - 'TST'
Traceback (most recent call last):
  File "/cds/home/z/zlentz/pydev/hutch_python/utils.py", line 60, in safe_load
    yield
  File "/cds/home/z/zlentz/pydev/hutch_python/load_conf.py", line 443, in load_conf
    happi_objs = get_happi_objs(db, lc, hutch)
  File "/cds/home/z/zlentz/pydev/hutch_python/happi.py", line 68, in get_happi_objs
    paths = light_ctrl.beamlines[endstation.upper()]
KeyError: 'TST'
```

Reply:
Do we need a special case here for when the user asks for an... interesting path?

Reply:
Maybe up at line 59 we dip into there for invalid beamline names?